packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -369,6 +369,12 @@ class ScheduleManagerCore {
      public afterOpProcessing(sequenceNumber: number) {
         assert(!this.localPaused, "can't have op processing paused if we are processing an op");
 
+        // If the inbound queue is ever empty, nothing to do!
+        if (this.deltaManager.inbound.length === 0) {
+            assert(this.pauseSequenceNumber === undefined, "there should be no pending batch if we have no ops");
+            return;
+        }
+
         // do we have incomplete batch to worry about?
         if (this.pauseSequenceNumber !== undefined) {
```

Comment:
Can we add a test for the scenario where we push non-batch messages followed by an incomplete batch and make sure that non-batch messages get processed? (Maybe you did and I missed it)

Reply:
Sure, will add.